## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 412566`

Logs: https://github.com/Emin017/nixpkgs-review-gha/actions/runs/15360337655


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ieda</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>ieda</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>ieda</summary>
<pre>PGNetlist.cc:(.text+0x1b30): undefined reference to `create_pg_netlist'
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: PGNetlist.cc:(.text+0x1b4c): undefined reference to `create_pg_node'
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: PGNetlist.cc:(.text+0x1b74): undefined reference to `create_pg_edge'
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: lib/libpg-netlist.a(PGNetlist.cc.o): in function `iir::IRPGNetlistBuilder::createRustRCData()':
PGNetlist.cc:(.text+0x1ca8): undefined reference to `create_rc_data'
collect2: error: ld returned 1 exit status
[1174/1176] Linking CXX shared module /nix/store/a6azrrn7ih6vgksqj8p003nvxrdg2wy9-iEDA-0-unstable-2025-05-30/bin/ista_cpp.cpython-312-aarch64-linux-gnu.so
lto-wrapper: warning: using serial compilation of 2 LTRANS jobs
lto-wrapper: note: see the '-flto' option documentation for more information
[1175/1176] Building CXX object src/operation/iNO/test/CMakeFiles/test_run_no.dir/run_no.cpp.o
In file included from /nix/store/nnsfrpjjdfj1g8ydgv6g4lxhlxkqwalb-gcc-14.2.1.20250322/include/c++/14.2.1.20250322/aarch64-unknown-linux-gnu/bits/os_defines.h:39,
                 from /nix/store/nnsfrpjjdfj1g8ydgv6g4lxhlxkqwalb-gcc-14.2.1.20250322/include/c++/14.2.1.20250322/aarch64-unknown-linux-gnu/bits/c++config.h:683,
                 from /nix/store/nnsfrpjjdfj1g8ydgv6g4lxhlxkqwalb-gcc-14.2.1.20250322/include/c++/14.2.1.20250322/bits/version.h:49,
                 from /nix/store/nnsfrpjjdfj1g8ydgv6g4lxhlxkqwalb-gcc-14.2.1.20250322/include/c++/14.2.1.20250322/any:35,
                 from /build/iEDA-src-2025-05-30/src/operation/iNO/api/NoApi.hpp:19,
                 from /build/iEDA-src-2025-05-30/src/operation/iNO/test/run_no.cpp:17:
/nix/store/csm3b8klpnl4748bpqi4wg3nsc81bly4-glibc-2.40-66-dev/include/features.h:422:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  422 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
ninja: build stopped: subcommand failed.</pre>
</details>
</details>

--- a/pkgs/by-name/ie/ieda/package.nix
+++ b/pkgs/by-name/ie/ieda/package.nix
@@ -35,11 +35,11 @@ let
   });
   rootSrc = stdenv.mkDerivation {
     pname = "iEDA-src";
-    version = "2025-04-14";
+    version = "2025-05-30";
     src = fetchgit {
       url = "https://gitee.com/oscc-project/iEDA";
-      rev = "51d198884cde2ecda643071a1a6cb4ec0e09d881";
-      sha256 = "sha256-kDVEAttSqa8l7qcRs7MQiBgPbAKBExEQvIE8tc7PLpM=";
+      rev = "3096147fcea491c381da2928be6fb5a12c2d97b7";
+      sha256 = "sha256-rPkcE+QFMlEuwwJ/QBgyLTXP5lWLQPj5SOlZysJ6WTI=";
     };
 
     patches = [
@@ -59,6 +59,11 @@ let
       })
     ];
 
+    postPatch = ''
+      # Comment out the iCTS test cases that will fail due to some linking issues on aarch64-linux
+      sed -i '17,28s/^/# /' src/operation/iCTS/test/CMakeLists.txt
+    '';
+
     dontBuild = true;
     dontFixup = true;
     installPhase = ''
@@ -71,7 +76,7 @@ let
 in
 stdenv.mkDerivation {
   pname = "iEDA";
-  version = "0-unstable-2025-04-14";
+  version = "0-unstable-2025-05-30";
 
   src = rootSrc;
 

